src/Widget/JavascriptResponse.php

```diff
@@ -58,7 +58,8 @@ public function __construct(Request $request, RendererInterface $renderer, $cont
      */
     private function renderContent($content)
     {
-        return 'document.write("' . trim(preg_replace('~[\r\n]+~', ' ', addslashes($content))) . '");';
+        $widgetWrapperId = "uit-widget";
+        return 'if(document.getElementById("'. $widgetWrapperId .'")){document.getElementById("'. $widgetWrapperId .'").innerHTML = "' . addslashes($content) . '";}else{document.write("' . trim(preg_replace('~[\r\n]+~', ' ', addslashes($content))) . '");}';
```

Comment:
Is the `trim(preg_replace(...))` part not necessary when using `innerHTML`?

Reply:
Jup added it

Reply:
The solution is good and the code is okay like this, but it would have been helpful if it was put in a variable first with some comment as to what it does/why it's necessary 😉 

For example:
```
// Explanation here...
$content = trim(preg_replace('~[\r\n]+~', ' ', addslashes($content)));
$widgetWrapperId = "uit-widget";
return 'if(document.getElementById("'. $widgetWrapperId .'")){document.getElementById("'. $widgetWrapperId .'").innerHTML = "' . $content . '";}else{document.write("' . $content . '");}';
```

But it's just a tip for the future. It's probably not worth it to change this yet again now.